### PR TITLE
MM-44926 Gekidou Settings Snackbars

### PR DESCRIPTION
**Pull request by avinashlng1080**
#### Summary
After each setting is saved, we are now showing a snackbar to let the user know that the operation was a success or not.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44926

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/hooks/why_did_you_update.ts [hunk 1]
```diff
@@ -30,7 +32,7 @@ export function useWhyDidYouUpdate(name: string, props: any) {
             // If changesObj not empty then output to console
             if (Object.keys(changesObj).length) {
                 // eslint-disable-next-line no-console
-                console.log('[why-did-you-update]', name, changesObj);
+                logInfo('[why-did-you-update]', name, changesObj);
```
Comment (larkox) [inline app/hooks/why_did_you_update.ts:35]: This was intended to be left as a console.log, since it shouldn't get to production, so 0/5 on whether we should have it as console.log or a log function.

But if we go with the log function, for sure is logDebug and you should remove the eslint disable comment.

File: app/products/calls/connection/connection.ts [hunk 2]
```diff
@@ -36,7 +37,7 @@ export async function newConnection(serverUrl: string, channelID: string, closeC
         voiceTrack.enabled = false;
         streams.push(stream);
     } catch (err) {
-        console.log('Unable to get media device:', err); // eslint-disable-line no-console
+        logError('Unable to get media device:', err); // eslint-disable-line no-console
```
Comment (larkox) [inline app/products/calls/connection/connection.ts:40]: Remove all the eslint disable comments in these changes.

File: app/screens/settings/config.ts [hunk 3]
```diff
@@ -1,7 +1,10 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {SNACK_BAR_TYPE} from '@constants/snack_bar';
```
Comment (larkox) [inline app/screens/settings/config.ts:4]: 0/5 the content of this file seems a bit out of place (I cannot wrap my head around it being in the screens folder, with a file name of config). Not a big deal, and can stay here, but I wanted to flag it.

File: app/screens/settings/display_clock/display_clock.tsx [hunk 4]
```diff
@@ -52,8 +52,8 @@ const DisplayClock = ({componentId, currentUserId, hasMilitaryTimeFormat}: Displ
             value: `${isMilitaryTimeFormat}`,
         };
 
-        savePreference(serverUrl, [timePreference]);
-        close();
+        const updatedSettings = savePreference(serverUrl, [timePreference]);
+        updateSettings(componentId, updatedSettings);
```
Comment (larkox) [inline app/screens/settings/display_clock/display_clock.tsx:56]: `updateSettings` will wait for the promise to return before closing. Previously, we were closing right away. Is this change intended/desired?

(Similar in other files)

**Review by larkox — CHANGES_REQUESTED** [review]
Some minor changes, but LGTM in general.

**Comment by matthewbirtch** [comment]
Based on [this conversation](https://community.mattermost.com/core/pl/fsfa9cnxaig778uzfkwrxkokth), I think we aren't going to proceed with the use of snackbars for settings

**Comment by avinashlng1080** [comment]
Alrighty - I'll be closing this PR as this functionality is not needed at the moment.
